activerecord/lib/active_record.rb

```diff
@@ -447,6 +447,13 @@ def self.suppress_multiple_database_warning=(value)
   singleton_class.attr_accessor :yaml_column_permitted_classes
   self.yaml_column_permitted_classes = [Symbol]
 
+  ##
+  # :singleton-method:
+  # Controls when to generate a value for <tt>has_secure_token</tt>
+  # declarations. Defaults to <tt>:initialize</tt>.
+  singleton_class.attr_accessor :generate_secure_token_on
+  self.generate_secure_token_on = :initialize
```

Comment:
If the goal is to change the default value for Rails 7.1, then the value here should be `:create`